### PR TITLE
Collaboration Taxonomy - New FF + New Page

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Preliminary housekeeping tasks for 1/3 sprint https://github.com/PostHog/product-internal/pull/244.

Note to reviewers: The large diff is scary! But it's not as bad as it looks. Most of the changes come from scenes, urls, and routing having an overall large blast radius when variables are renamed. Feel free to smoke test most of the PR to save time. Everything is behind an FF.

As part of starting taxonomy work, we've decided to split out live events into its separate page and dedicate a separate "Taxonomy" page. This PR creates a fork of today's components into separate folders and components and prefixes old parts with `LEGACY_` for easier future removal.

- [x] Introduce new `collaborations-taxonomy` FF
- [x] Move events tab to new Live Events page
- [x] Rename "Events & Actions" -> "Taxonomy"
- [x] Move events tab into separate "Live Events" page
- [x] Rename "Actions" -> "Calculated Events"
- [x] Persist event/property id to url so that url's can be linked directly

TODO in future PR's
- [ ] Consolidate "Properties Stats" tab into "Events Stats"
- [ ] Icons for new pages - "Taxonomy" and "Live Events" @clarkus 
- [ ] Implement Taxonomy page

**Without FF**

Screen Shot 2022-01-10 at 5 24 13 PM

**With FF**
**/events_v2**

Screen Shot 2022-01-10 at 5 24 26 PM

**/taxonomy**

Screen Shot 2022-01-10 at 5 24 31 PM

## How did you test this code?

This should be a harmless PR. Without the new FF, the app does not change in how it looks or functions.

### CODE REVIEW

File: frontend/src/lib/components/CommandPalette/commandPaletteLogic.ts [hunk 1]
```diff
@@ -743,6 +764,8 @@ export const commandPaletteLogic = kea<
                 },
             }
 
+            console.log('GO TO', goTo, featureFlagLogic[FEATURE_FLAGS.COLLABORATIONS_TAXONOMY])
+
```
Comment (mariusandra) [inline frontend/src/lib/components/CommandPalette/commandPaletteLogic.ts:768]: `console.log`

**Review by mariusandra — APPROVED** [review]
Looks like a good base to extend this from! Found a `console.log` in the code (and removed it), and another small issue with the events page, which is probably out of scope here, but documenting just in case:

**Comment by mariusandra** [comment]
Two tests are consistently failing though, both with:
```
      cy:command ✘  get	[data-attr="menu-item-events"]
```

File: frontend/src/layout/navigation/SideBar/SideBar.tsx [hunk 2]
```diff
@@ -204,7 +204,14 @@ function Pages(): JSX.Element {
                 <PageButton icon={<IconExperiment />} identifier={Scene.Experiments} to={urls.experiments()} />
             )}
             <LemonSpacer />
-            <PageButton icon={<IconGroupedEvents />} identifier={Scene.Events} to={urls.events()} />
+            {featureFlags[FEATURE_FLAGS.COLLABORATIONS_TAXONOMY] ? (
+                <>
+                    <PageButton icon={<IconGroupedEvents />} identifier={Scene.Events} to={urls.events()} />
+                    <PageButton icon={<IconGroupedEvents />} identifier={Scene.Taxonomy} to={urls.taxonomy()} />
```
Comment (paolodamico) [inline frontend/src/layout/navigation/SideBar/SideBar.tsx:210]: Feel free to tackle in a separate PR, but not sure "Taxonomy" is the right word from a user's perspective (is it clear enough?), also while the concept does encompass Actions, I'm not sure anyone looking for actions will think of looking here.

File: frontend/src/scenes/urls.ts [hunk 3]
```diff
@@ -7,12 +7,20 @@ export const urls = {
...
-    events: () => '/events',
+    // TODO: Deprecate below with `collaborations-taxonomy` FF
+    LEGACY_createAction: () => `/action`, // TODO: For consistency, this should be `/action/new`
+    LEGACY_action: (id: string | number) => `/action/${id}`,
+    LEGACY_actions: () => '/events/actions',
+    LEGACY_eventStats: () => '/events/stats',
+    LEGACY_eventPropertyStats: () => '/events/properties',
+    LEGACY_events: () => '/events',
+    // TODO: Deprecate above with `collaborations-taxonomy` FF
+    events: () => '/events_v2', // live events
```
Comment (paolodamico) [inline frontend/src/scenes/urls.ts:18]: Why not something like `live_events` to be consistent with the sidebar and page name.

File: frontend/src/scenes/urls.ts [hunk 4]
```diff
@@ -7,12 +7,20 @@ export const urls = {
...
+    taxonomy: () => '/taxonomy',
+    taxonomyEventStats: () => '/taxonomy/events/stats',
```
Comment (paolodamico) [inline frontend/src/scenes/urls.ts:20]: Not sure how you're thinking about the new taxonomy page, but I would've assumed that stats are part of the new taxonomy and then you just have events and properties tabs (and maybe actions?)

File: frontend/src/scenes/urls.ts [hunk 5]
```diff
@@ -7,12 +7,20 @@ export const urls = {
...
+    taxonomyEventStatsView: (id: string | number) => `/taxonomy/events/stats/${id}`,
+    taxonomyCalculatedEvents: () => '/taxonomy/calculated_events',
```
Comment (paolodamico) [inline frontend/src/scenes/urls.ts:22]: What is a calculated event? If we're renaming actions I think we should tackle this separately as it'll introduce significant confusion for existing users.

**Review by paolodamico — DISMISSED** [review]
The refactoring of the legacy stuff lgtm, though I'm a bit unclear on how the new stuff will work, added some comments inline.

File: frontend/src/scenes/urls.ts (hunk 4 above)
Comment (alexkim205) [reply frontend/src/scenes/urls.ts:20]: Yup that's completely right and partly my fault for being bad at naming. Tomorrow's taxonomy page will have two tabs called Events and Actions (Calculated Events in the future). Properties will be brought under Events tab (as discussed in https://github.com/PostHog/posthog/issues/7589) and stats for both will be incorporated into the taxonomy definitions UX.

I'll rename this from `taxonomyEventStats` to `taxonomyEvent` to communicate this better.

File: frontend/src/scenes/urls.ts (hunk 5 above)
Comment (alexkim205) [reply frontend/src/scenes/urls.ts:22]: Yes Rick and I were talking about how actions is a misleading term that makes it hard to distinguish from what an event is. But totally agree, let's tackle this renaming thing in a separate initiative.

File: frontend/src/layout/navigation/SideBar/SideBar.tsx (hunk 2 above)
Comment (alexkim205) [reply frontend/src/layout/navigation/SideBar/SideBar.tsx:210]: I was struggling w naming this one. Maybe "Events & Actions Taxonomy/Definitions" but doesn't quite roll off the tongue. I'll keep this now and iterate on it later.

**Review by clarkus — DISMISSED** [review]
I don't think the feature name "live events" is meaningful. All events are live, we just don't show them that way. Likewise, taxonomy, while technically correct, isn't a super approachable term. Why are we renaming actions? This is documented as part of the solution, but how does it add value? What problem is being solved with that change? 

Proposal - leave the "Events & Actions" feature name as is. Add a new feature called "data management" or something much more targeted. Thoughts?

**Review by clarkus — COMMENTED** [review]
How does a user create a new action / calculated event? I don't see any actions for that in the screenshots yet.

**Comment by alexkim205** [comment]
> I don't think the feature name "live events" is meaningful. All events are live, we just don't show them that way. Likewise, taxonomy, while technically correct, isn't a super approachable term. 

👍 That makes sense. I'll rename live events to Events and Actions and Taxonomy to Data Management. Also moved data management under toolbar since it's a setting that can be changed project wide.

> Why are we renaming actions? This is documented as part of the solution, but how does it add value? What problem is being solved with that change?

The problem we're trying to solve is that it's difficult to get across to a new (and returning users) the difference between an action and an event. In my experience onboarding, this contributed to a steeper learning curve since it wasn't a term that is commonly used outside of PH in other product analytics spaces (Both He@p and M1xp@nel stick to this concept of events and defined events). Internally it seems we're also aware of the learning gap, since we have this uninspiring blurb to describe what an action can be.

Screen Shot 2022-01-11 at 10 49 51 AM

With that being said though, I'm not sure if renaming actions to something more descriptive like calculated events is the right solution here. It might just be a matter of communicating this to the user in a clearer way.

> How does a user create a new action / calculated event? I don't see any actions for that in the screenshots yet.

This isn't implemented yet. This PR just lays the groundwork for routing, but these new pages will be coming in future PRs soon.

File: frontend/src/scenes/urls.ts (hunk 4 above)
Comment (paolodamico) [reply frontend/src/scenes/urls.ts]: That makes sense, but then I would assume `/taxonomy` page just loads the events tab with all that stuff? and then you just have an additional page for actions/calculated events

**Comment by Twixes** [comment]
I was debugging a customer problem and was really confused why my changes to EventsTable.tsx are taking no effect, but just noticed that the reason is now there being TWO practically the same `EventsTable.tsx`s – just that one in `LEGACY_events` and the other in `events`. To be honest I don't totally see why EventsTable would be duplicated at this stage, as the current tweaks are microscopic, but assuming it's really necessary to duplicate these hundreds of lines of codes, in the future please rename the file names too.

**Pull request merged**